src/cpp/desktop/DesktopGwtWindow.cpp

```diff
@@ -118,6 +118,22 @@ bool GwtWindow::event(QEvent* pEvent)
    return BrowserWindow::event(pEvent);
 }
 
+void GwtWindow::onCloseWindowShortcut()
+{
+   // check to see if the window has desktop hooks (not all GWT windows do); if it does, check to
+   // see whether it has a closeSourceDoc() command we should be executing instead
+   webPage()->runJavaScript(
+            QStringLiteral(
+               "if (window.desktopHooks) "
```

Comment:
Worth noting that writing this kind of inline code is a lot nicer with C++11 raw string literals